### DISCUSSION TITLE
Get Screenposition in Blazer

### DISCUSSION THREAD

**Discussion by peder1896** [Mapsui/Mapsui#1963] [Q&A]
Hello 
With Mapsui 3.0.2 I do like this in Xamarin.Forms to get screenposition
```csharp
public (double screenX, double screenY) calcScreenPosFromLatLong(double lat, double lon, double scale, SKSize svgSize)
{
     double screenX = 0, screenY = 0;
     var sphericalMercatorCoordinateCenter = SphericalMercator.FromLonLat(centerOfRadarLonLat.X, centerOfRadarLonLat.Y);
     mapControl.Navigator.NavigateTo(sphericalMercatorCoordinateCenter, scale);
     var sphericalMercatorCoordinate1 = SphericalMercator.FromLonLat(lon, lat);
     Mapsui.Geometries.Point p2 = mapControl.Viewport.WorldToScreen(sphericalMercatorCoordinate1);
     return ((p2.X + 512) - (svgSize.Width / 2), (p2.Y + 512) - (svgSize.Height / 2));
 }
```
how do i migrate this to Mapsui Blazor 4.0.0-beta.10
Thanks Peder

**Comment by pauldendulk** [comment]
Viewport.WorldToScreen is now an extension method, you need to add `using Mapsui.Extensions`.

The Navigator has moved to the Map and most of its methods are renamed. Now it is: `mapControl.Map.Navigator.CenterOnAndZoomTo(sphericalMercatorCoordinateCenter, scale);`

Does this answer your question?

**Comment by pilotshamn** [comment]
Thank you for you answer sorry for late reply but i am on holiday will test as soon as i am back.
